moonraker/components/file_manager/file_manager.py

```diff
@@ -1537,18 +1549,40 @@ async def _process_metadata_update(self) -> None:
 
     async def _run_extract_metadata(self,
                                     filename: str,
-                                    ufp_path: Optional[str]
+                                    ufp_path: Optional[str],
+                                    gcode_path: Optional[str]
                                     ) -> None:
         # Escape single quotes in the file name so that it may be
         # properly loaded
         filename = filename.replace("\"", "\\\"")
         cmd = " ".join([sys.executable, METADATA_SCRIPT, "-p",
                         self.gc_path, "-f", f"\"{filename}\""])
         timeout = 10.
+        if gcode_path is not None and os.path.isfile(gcode_path):
+            gcode_path.replace("\"", "\\\"")
+            cmd += f" -g \"{gcode_path}\""
+
         if ufp_path is not None and os.path.isfile(ufp_path):
             timeout = 300.
             ufp_path.replace("\"", "\\\"")
             cmd += f" -u \"{ufp_path}\""
+
+        exclude_object_enabled = False
+        try:
+            kapis: APIComp = self.server.lookup_component('klippy_apis')
+            kobjects = await kapis.get_object_list(default=None)
+            if kobjects is not None:
+                exclude_object_enabled = "exclude_object" in kobjects
+        except Exception:
+            # After logging, continue processing with the assumption that
+            # exclude_object is not enabled.
+            logging.exception(
+                "Error querering klipper for exclude_object status")
+
+        if exclude_object_enabled:
+            timeout = 300.
+            cmd += " --exclude-object"
```

Comment:
Rather than attempt to detect if `exclude_object` is configure in Klipper I think we should just add a configuration option to the `file_manager`, ie `enable_object_processing`.  It is perfectly acceptable for a user to upload a file when Klipper in not available and I think we want to be able to process objects in that scenario.